ef-validation/README.md

```diff
@@ -1,3 +1,279 @@
 # Entity Framework Validation
 
+## Validierung allgemein 
+
+* Bietet eine Möglichkeit, eine benutzerdefinierte Regel zu erstellen, um die Gültigkeit von Benutzereingaben zu überprüfen.
+
+* Selbes Konzept für mehrere Ebenen:
+    * WPF -> Validierung am Desktop
+    * Web-UI clientseitig -> Validierung im Browser
+    * Entity direkt -> Domainspezifisch
+    * Persistenzschicht -> Datenbankabhängig
+
+* Validation-Framework:
+    * Ermöglicht Validierung in allen Bereichen
+    * ValidationExecption zum Transport von Exceptions 
+    * Kernklasse Validator (statisch):
+        * ValidateObject(Object, ValidationConext)
+        * TryValidateObject(Object, ValidationContext ICollection <ValidationResult<x>>)
+        * ValidateProperty(Object, ValidationContext)
+    
+## Möglichkeiten
+* DataAnnotations für einzelne Properties:
+    * MaxLength, RegEx, EmailAddress, ...
+    * CustomValidation
+    * Eigene Annotationsklassen
+
+
+### Beispiel für Vordefinierte Annotations:
+* Required, Range
+```c#
+[Required]
+[Range(1, int.MaxValue, ErrorMessage = "Runde muss positive Ganzzahl > 0 sein")]
+public int Round {get; set;}
+
+```
+* Required, MinLength, MaxLength
+```c#
+private string _editName;
+ 
+[Required]
+[MinLength(1, ErrorMessage = "Produktname muss mindestens 1 Zeichen lang sein")]
+[MaxLength(20, ErrorMessage = "Produktname darf maximal 20 Zeichen lang sein")]
+public string EditName
+{
+    get => _editName;
+    set
+    {
+        _editName = value;
+        OnPropertyNameChanged();
+    }
+}
+```
+![](ValidationProduct.png)
+
+### Beispiel Custom Validation:
+```c#
+public class PhoneValidator : ValidatonAttribute
+{
+    public override bool IsValid(object value)
+    {
+        return new RegularExpressionAttribute(@"^[+0]\d+").IsValid(Convert.ToString(value).Trim());
```

Comment:
Dieses Beispiel ist nicht optimal - es gibt ja ein eigenes [RegularExpressionAttribute ](https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.regularexpressionattribute?view=net-5.0) welche Regular Expressions validiert.

Sie könnten in diesem Beispiel eine SVNr-Validierung implementieren. Den Beispielcode der Validierung müssen sie für das Verständnis nicht komplett ausimplementieren.